module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,56 @@
+#include "compression.hpp"
+
+#include <iostream>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& bitmap) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressed;
+    for (const auto& line : bitmap) {
+        uint8_t count{};
+        uint8_t pixel = line.at(0);
+        for (const auto& el : line) {
+            if (pixel == el) {
+                ++count;
+            } else {
+                compressed.push_back(std::make_pair(pixel, count));
+                count = 1;
+                pixel = el;
+            }
+        }
+        compressed.push_back(std::make_pair(pixel, count));
+    }
+
+    return compressed;
+}
+
+std::array<std::array<uint8_t, width>, height> decompressGrayscale(const std::vector<std::pair<uint8_t, uint8_t>>& compressed) {
+    std::array<std::array<uint8_t, width>, height> bitmap;
+
+    size_t count{}, line{}, el{};
+    for (const auto& pair : compressed) {
+        if (count >= width) {
+            count = 0;
+            el = 0;
+            ++line;
+        }
```

Comment:
You don't need 3 variables. Try to use iterator. You will simplify code, and make it more readable.

Reply:
Iterator used finally! Thank you for not approving it that long MA! While working on STL#2 I had to use iterators and implemented one here as well. Indeed it's more readable now.